controllers/main.go

```diff
@@ -185,6 +207,45 @@ func (controller *MainController) getNetworkName() string {
 	return controller.params.Name
 }
 
+// agendas returns agendas and their statuses. Fetches agenda status from
+// dcrdata.org if past agenda.Timer limit from previous fetch. Caches agenda
+// data for agendasCacheLife. This method is safe for concurrent use.
+func (controller *MainController) agendas() []agenda {
+	agendasCache.Lock()
+	defer agendasCache.Unlock()
+	now := time.Now()
+	if agendasCache.timer.After(now) {
+		return *agendasCache.agendas
+	}
+	agendasCache.timer = now.Add(agendasCacheLife)
+	url := fmt.Sprintf("https://%s.dcrdata.org/api/agendas", controller.getNetworkName())
+	agendaInfos, err := dcrDataAgendas(url)
+	if err != nil {
+		// Ensure the next call tries to fetch statuses again.
+		agendasCache.timer = time.Time{}
+		log.Warnf("unable to retrieve data from %v: %v", url, err)
+		// If we have initialized agendas, return that.
+		if agendasCache.agendas != nil {
+			return *agendasCache.agendas
```

Comment:
If an error occurs we should always return here, regardless of what is cached. All of the code beyond this point assumes a successful result has been returned from dcrdata, but it hasnt.

Reply:
I wrote it so that an error here will perform exactly like it does now. We still have all the  agenda info except for statuses. If this fails it will use cache if available, and not show statuses if not.

Reply:
If you could try changing the url to something silly, you'll see it log an error and show the agendas as they are now.

Reply:
Tested thoroughly, you are correct. Its different to how I would have done it, but it works and that is what matters :+1:  Approving